feat : Add IPv6 multiaddress

Harshil-Jani opened the pull request:
Solving #547 
- [x] Adding Listener to Internet connection 
- [x] Adding Listener to Lan connection
- [x] Make `listen` configurable with IPv4 and IPv6.
- [x] Automatic procedure for clients to upgrade to new configuration.

Harshil-Jani committed:
feat : Add IPv6 multiaddress

Signed-off-by: Harshil Jani <harshiljani2002@gmail.com>

Harshil-Jani committed:
feat: Make LAN and Internet configurable with IPv6

Signed-off-by: Harshil Jani <harshiljani2002@gmail.com>

Harshil-Jani commented:
IPv4 and IPv6 MultiAddresses are now configured for both LAN and Internet. What we would do for writing an automatic upgrade procedure ?  I was reading some answers over here https://stackoverflow.com/questions/232347/how-should-i-implement-an-auto-updater and I think we would need to check if the configuration file now has IPv6 added to it or not and based on it we can run our script to update. 

Please let me know what could be a possible solution for this.

MathJud commented:
libqaul has an own [upgrade module in `libqaul/src/utilities/upgrade`](https://github.com/qaul/qaul.net/tree/main/rust/libqaul/src/utilities/upgrade), which is executed before loading the configuration, as a wrong file structure of the configuration results in a panic.
The decision of the upgrade procedure is bound to the library version.
We can bind it to the next version `beta 15`.

Each upgrade procedure is contained in an own folder. You can take the [upgrade to beta 9](https://github.com/qaul/qaul.net/tree/main/rust/libqaul/src/utilities/upgrade/v2_0_0_beta_9) as a reference.

MathJud committed:
Made the amount of interfaces configurable via config file.

The amount of listening interfaces are flexibly configurable in the configuration file.

MathJud commented:
The IPv6 works good, I just made the amount of interfaces per connection module dynamically configurable by using a for loop going through the configuration array.

Harshil-Jani committed:
feat: Writing an automatic upgrade module for beta-15

We have had a change for the configuration of Internet and Lan for IPv6.
So this commit has a upgrade module written for users who would
automatically be upgraded to the beta-15 version which includes IPv6.

Co-authored-by: Mathias Jud <MathJud@users.noreply.github.com>
Signed-off-by: Harshil Jani <harshiljani2002@gmail.com>

MathJud committed:
fixed upgrade peer name copy in config.yml
